src/shared_realm.cpp

```diff
@@ -30,7 +30,12 @@
 
 #include "util/format.hpp"
 
+#if REALM_VER_MAJOR < 2
```

Comment:
In these situations I typically prefer to structure the condition so that the code for the new version is first, with the older code second. It makes it a little easier to skim just the _current_ code.